analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TooManyGenericParameters.CSharp10.cs

```diff
@@ -1,22 +1,21 @@
-﻿void LocalFoo<T1, T2, T3>() { }
-void LocalBar<T1, T2, T3, T4>() { } // Noncompliant {{Reduce the number of generic parameters in the '.LocalBar' method to no more than the 3 authorized.}}
+﻿namespace Tests.Diagnostics;
```

Comment:
Namespaces are usually just noise for the tests and we don't need to add them. Unless they serve a purpose

Reply:
This one kind of does. It requires `WithConcurrentAnalysis(false)` in the test. I'm not sure if it should be removed or not. What do you think?

Reply:
We do some [ugly magic](https://github.com/SonarSource/sonar-dotnet/blob/ec9a976fe885c7cb663eb23d7e6990607ff5f8cd/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs#L183-L194) to craft a 2nd file with different namespace and duplicated content in order to increase changes to catch issues with concurrency. 

If we have to choose between namespace in test file and `WithConcurrentAnalysis(false)`, then the namespace in test file is more valuable.

Reply:
..currently we have both, that's confusing. One of them should / can go away

Reply:
I would keep it because it is a new language feature.